Dockerfile.scratch

```diff
@@ -0,0 +1,48 @@
+# Build stage
+ARG goversion
+FROM golang:${goversion}-alpine as builder
+RUN apk add build-base git mercurial ca-certificates upx
+RUN apk add --update gcc musl-dev
+ADD go.mod /spire/go.mod
+ADD proto/spire/go.mod /spire/proto/spire/go.mod
+RUN cd /spire && go mod download
+ADD . /spire
+WORKDIR /spire
+RUN make build-static
+
+# UPX is a free, portable, extendable, high-performance executable packer for several executable formats.
+RUN upx ./bin/spire-server-static \
```

Comment:
I would prefer we not introduce this tool in this PR. I would rather see either both the scratch and Alpine images using binaries compressed by this tool or neither of them. IMO introducing this `upx` tool should be a different PR.

What was the intent of using `upx` here? To reduce the size of the Docker images? I did notice in my local testing that the `spire-server-scratch` image was ~16.7 MB using `upx` and ~62.1 MB without, so I do think it's worth considering using this.

Reply:
Ok, the intent was to reduce the size a component (mostly spire-agent) that potentially is going to be in use in 100's of containers.  It probably has no effect on memory, but makes a difference on startup time as the bytes have to move over the network or at least that was my thinking.